components/places/android/src/main/java/org/mozilla/places/PlacesConnection.kt

```diff
@@ -250,14 +299,34 @@ class SyncAuthInfo (
 )
 
 /**
- * An API for interacting with Places.
+ * An API for interacting with Places. This is the top-level entry-point, and
+ * exposes functions which return lower-level objects with the core
+ * functionality.
  */
-interface PlacesAPI {
+interface PlacesApiInterface {
+    fun open_reader(): ReadablePlacesConnectionInterface
+
+    fun open_writer(): WritablePlacesConnectionInterface
+
     /**
-     * Record a visit to a URL, or update meta information about page URL. See [VisitObservation].
+     * Syncs the places stores.
+     *
+     * Note that this function blocks until the sync is complete, which may
+     * take some time due to the network etc. Because only 1 thread can be
+     * using a PlacesAPI at a time, it is recommended, but not enforced, that
+     * you have all connections you intend using open before calling this.
      */
-    fun noteObservation(data: VisitObservation)
+    fun sync(syncInfo: SyncAuthInfo)
+}
 
+interface PlacesConnectionInterface {
```

Comment:
I don't think it's common to put `Interface` in interface names. I've seen `IBlah` before (although that was in C#). In general these names are pretty verbose.

I guess that will be a question @grigoryk can help with when this stuff is ready for his review.